tests/h/views/client_test.py

```diff
@@ -19,6 +20,52 @@ def test_annotator_token_returns_token(generate_jwt, pyramid_request):
     assert result == generate_jwt.return_value
 
 
+@pytest.mark.usefixtures('routes')
+class TestRenderApp(object):
+    def test_uses_client_boot_script(self, pyramid_request_):
+        pyramid_request_.feature.flags['use_client_boot_script'] = True
+        rsp = client.render_app(pyramid_request_)
+        assert '<script src="http://example.com/assets/client/boot.js"></script>' in rsp.text
+
+    def test_does_not_use_client_boot_script_when_disabled(self, pyramid_request_):
+        pyramid_request_.feature.flags['use_client_boot_script'] = False
+        rsp = client.render_app(pyramid_request_)
+        assert '<script src="http://example.com/assets/client/boot.js"></script>' not in rsp.text
+
+
+@pytest.mark.usefixtures('routes')
+class TestEmbed(object):
+    def test_uses_client_boot_script(self, pyramid_request_):
+        pyramid_request_.feature.flags['use_client_boot_script'] = True
+        ctx = client.embed(pyramid_request_)
+        assert ctx['client_boot_url'] == 'http://example.com/assets/client/boot.js'
+
+    def test_does_not_use_client_boot_script_when_disabled(self, pyramid_request_):
+        pyramid_request_.feature.flags['use_client_boot_script'] = False
+        ctx = client.embed(pyramid_request_)
+        assert ctx['client_boot_url'] is None
+
+    def test_sets_content_type(self, pyramid_request_):
+        client.embed(pyramid_request_)
+        assert pyramid_request_.response.content_type == 'text/javascript'
+
+
+@pytest.fixture
+def pyramid_request_(pyramid_request):
```

Comment:
This fixture needs to be named differently than the `pyramid_request` fixture which it extends, as otherwise the registry gets reset to a new object before individual test cases actually run.